### PR TITLE
Fix blob context when db_iter uses seek

### PR DESCRIPTION
**Pull request by tabokie**
Fix: when `db_iter` falls back to using seek by `FindValueForCurrentKeyUsingSeek`, `is_blob_` flag is not properly set on encountering BlobIndex.
Also patch existing test for the mentioned code path.
Signed-off-by: tabokie <xy.tao@outlook.com>

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
@ltamasi PTAL. This is a BlobDB bug.

File: db/db_blob_index_test.cc [hunk 1]
```diff
@@ -398,6 +398,26 @@ TEST_F(DBBlobIndexTest, Iterate) {
     verify(15, Status::kOk, get_value(16, 0), get_value(14, 0),
            create_blob_iterator, check_is_blob(false));
 
+    // Iterator with blob support and using seek.
```
Comment (yiwu-arbug) [inline db/db_blob_index_test.cc]: how do you make sure `FindValueForCurrentKeyUsingSeek` is called in this test?

**Comment by yiwu-arbug** [comment]
@tabokie please also update HISTORY.md to mention the fix.

File: db/db_blob_index_test.cc (hunk 1 above)
Comment (tabokie) [reply db/db_blob_index_test.cc]: From this [line](https://github.com/facebook/rocksdb/blob/3b3b96d6933c64b9c47dba19020e0dcce214c668/db/db_iter.cc#L724), we know that when `max_skip_==0` backward iteration must call `FindValueForCurrentKeyUsingSeek` to retrieve value. I verify the case by running the test without the fix, and it failed.

**Pull request closed without merging**